tests/Fixtures/inc/Engine/CriticalPath/CriticalCSSSubscriber/asyncCss.php

```diff
@@ -0,0 +1,270 @@
+			],
+			'expected' => [
+				'html' => '<!doctype html><html lang="en-US"><head><meta charset="UTF-8" /><link rel="stylesheet"  as="style" type="text/css" onload="this.media=\'print\'" href="http://www.example.com/file1.css" media="print"></head><body>Content here<noscript><link rel="stylesheet" type="text/css" href="http://www.example.com/file1.css" media="print"></noscript></body></html>'
+			]
+		],
+
+		'shouldDeferCssFilesWithOnloadExists' => [
+			'config' => [
+				'constants' => [
+					'DONOTROCKETOPTIMIZE' => false,
+					'DONOTASYNCCSS'       => false
+				],
+				'options' => [
+					'async_css' => true
+				],
+				'exclude_options' => [
+					'async_css' => false
+				],
+				'html' => '<!doctype html><html lang="en-US"><head><meta charset="UTF-8" /><link onload="alert(\'Hi\')" rel="stylesheet" type="text/css" href="http://www.example.com/file1.css" media="print"></head><body>Content here</body></html>'
+			],
+			'expected' => [
+				'html' => '<!doctype html><html lang="en-US"><head><meta charset="UTF-8" /><link onload="this.media=\'print\'" rel="stylesheet"  as="style" type="text/css" href="http://www.example.com/file1.css" media="print"></head><body>Content here<noscript><link onload="alert(\'Hi\')" rel="stylesheet" type="text/css" href="http://www.example.com/file1.css" media="print"></noscript></body></html>'
+			]
+		],
+		'shouldDeferCssFilesWithOnloadExistsAndQuotes' => [
+			'config' => [
+				'constants' => [
+					'DONOTROCKETOPTIMIZE' => false,
+					'DONOTASYNCCSS'       => false
+				],
+				'options' => [
+					'async_css' => true
+				],
+				'exclude_options' => [
+					'async_css' => false
+				],
+				'html' => '<!doctype html><html lang="en-US"><head><meta charset="UTF-8" /><link onload=\'alert(\'Hi\')\' rel="stylesheet" type="text/css" href="http://www.example.com/file1.css" media="print"></head><body>Content here</body></html>'
+			],
+			'expected' => [
+				'html' => '<!doctype html><html lang="en-US"><head><meta charset="UTF-8" /><link onload="this.media=\'print\'" rel="stylesheet"  as="style" type="text/css" href="http://www.example.com/file1.css" media="print"></head><body>Content here<noscript><link onload=\'alert(\'Hi\')\' rel="stylesheet" type="text/css" href="http://www.example.com/file1.css" media="print"></noscript></body></html>'
+			]
+		],
+
+		'shouldDeferCssFilesWithOnloadExistsAtTheEnd' => [
+			'config' => [
+				'constants' => [
+					'DONOTROCKETOPTIMIZE' => false,
+					'DONOTASYNCCSS'       => false
+				],
+				'options' => [
+					'async_css' => true
+				],
+				'exclude_options' => [
+					'async_css' => false
+				],
+				'html' => '<!doctype html><html lang="en-US"><head><meta charset="UTF-8" /><link rel="stylesheet" type="text/css" href="http://www.example.com/file1.css" media="print" onload="alert(\'Hi\')"></head><body>Content here</body></html>'
```

Comment:
Question @crystinutzaa:

If the `<link>` already has `media="print"`, why do we need `onload="this.media='print'"? That attribute already exists in the element.

Reply:
If there is already `media="print"` then there should be also an onload="this.media='all'". Otherwise the CSS will not work. 
In case the original media is print I can add a check and switch it to all.
What do you think?